packages/config-plugins/src/android/Notifications.ts

```diff
@@ -0,0 +1,145 @@
+import { ExpoConfig } from '@expo/config-types';
+import { generateImageAsync } from '@expo/image-utils';
+import fs from 'fs-extra';
+import path from 'path';
+
+import { ConfigPlugin } from '../Plugin.types';
+import { createAndroidManifestPlugin } from '../plugins/android-plugins';
+import { withDangerousMod } from '../plugins/core-plugins';
+import * as Colors from './Colors';
+import { ANDROID_RES_PATH, dpiValues } from './Icon';
+import {
+  addMetaDataItemToMainApplication,
+  AndroidManifest,
+  getMainApplicationOrThrow,
+  removeMetaDataItemFromMainApplication,
+} from './Manifest';
+import { buildResourceItem, readResourcesXMLAsync } from './Resources';
+import { writeXMLAsync } from './XML';
+
+const BASELINE_PIXEL_SIZE = 24;
+export const META_DATA_NOTIFICATION_ICON = 'expo.modules.notifications.default_notification_icon';
+export const META_DATA_NOTIFICATION_ICON_COLOR =
+  'expo.modules.notifications.default_notification_color';
+export const NOTIFICATION_ICON = 'notification_icon';
+export const NOTIFICATION_ICON_RESOURCE = `@drawable/${NOTIFICATION_ICON}`;
+export const NOTIFICATION_ICON_COLOR = 'notification_icon_color';
+export const NOTIFICATION_ICON_COLOR_RESOURCE = `@color/${NOTIFICATION_ICON_COLOR}`;
+
+export const withNotificationIconColor: ConfigPlugin = config => {
+  return withDangerousMod(config, [
+    'android',
+    async config => {
+      await setNotificationIconColorAsync(config, config.modRequest.projectRoot);
+      return config;
+    },
+  ]);
+};
+
+export const withNotificationManifest = createAndroidManifestPlugin(
+  setNotificationConfigAsync,
+  'withNotificationManifest'
+);
+
+export function getNotificationIcon(config: ExpoConfig) {
+  return config.notification?.icon || null;
+}
+
+export function getNotificationColor(config: ExpoConfig) {
+  return config.notification?.color || null;
+}
+
+/**
+ * Applies configuration for expo-notifications, including
+ * the notification icon and notification color.
+ */
+export async function setNotificationIconAsync(config: ExpoConfig, projectRoot: string) {
```

Comment:
Is this used anywhere?

Reply:
woops, it's supposed to be! created the appropriate plug in